integration/fuse/src/main/java/alluxio/fuse/cli/command/MetadataCacheCommand.java

```diff
@@ -11,67 +11,82 @@
+import java.util.Map;
+import java.util.function.BiFunction;
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * Gets information of or makes changes to the Fuse client metadata cache.
  */
 @ThreadSafe
-public final class MetadataCacheCommand {
-  private static final Logger LOG = LoggerFactory.getLogger(MetadataCacheCommand.class);
+public final class MetadataCacheCommand extends AbstractFuseShellCommand {
 
-  private final AlluxioConfiguration mConf;
-  private final FileSystem mFileSystem;
+  private static final Map<String, BiFunction<FileSystem, AlluxioConfiguration,
+      ? extends FuseCommand>> SUB_COMMANDS = new HashMap<>();
 
-  public MetadataCacheCommand(FileSystem fileSystem, AlluxioConfiguration alluxioConfiguration) {
-    mFileSystem = fileSystem;
-    mConf = alluxioConfiguration;
+  static {
+    SUB_COMMANDS.put("dropAll", DropAllCommand::new);
+    SUB_COMMANDS.put("drop", DropCommand::new);
+    SUB_COMMANDS.put("size", SizeCommand::new);
   }
 
-  public URIStatus run(AlluxioURI path, String[] argv) throws InvalidArgumentException {
-    if (!mConf.getBoolean(PropertyKey.USER_METADATA_CACHE_ENABLED)) {
-      throw new UnsupportedOperationException(String
-          .format("metadatacache command is not supported when %s is false",
-              PropertyKey.USER_METADATA_CACHE_ENABLED.getName()));
-    }
-    if (argv.length > 1) {
-      throw new InvalidArgumentException("metadatacache requires 0 or 1 arguments");
-    }
-    if (argv.length == 0) {
-      return getMetadataCacheSize();
-    }
-    String cmd = argv[0];
-    switch (cmd) {
-      // TODO(lu) better organizing sub command names
-      case "dropAll":
-        ((MetadataCachingBaseFileSystem) mFileSystem).dropMetadataCacheAll();
-        return new URIStatus(new FileInfo().setCompleted(true));
-      case "drop":
-        ((MetadataCachingBaseFileSystem) mFileSystem).dropMetadataCache(path);
-        return new URIStatus(new FileInfo().setCompleted(true));
-      case "size":
-        return getMetadataCacheSize();
-      default:
-        throw new InvalidArgumentException(String
-            .format("%s is not a valid subcommand for command metadatacache", cmd));
+  private Map<String, FuseCommand> mSubCommands = new HashMap<>();
+
+  /**
+   * @param fs filesystem instance from fuse command
+   * @param conf configuration instance from fuse command
+   */
+  public MetadataCacheCommand(FileSystem fs, AlluxioConfiguration conf) {
+    super(fs, conf);
+    SUB_COMMANDS.forEach((name, constructor) -> {
+      mSubCommands.put(name, constructor.apply(fs, conf));
+    });
+  }
+
+  @Override
+  public boolean hasSubCommand() {
+    return true;
+  }
+
+  @Override
+  public Map<String, FuseCommand> getSubCommands() {
+    return mSubCommands;
+  }
+
+  @Override
+  public String getCommandName() {
+    return "metadatacache";
+  }
+
+  @Override
+  public String getUsage() {
+    // Show usage: metadatacache.([drop] [size] [dropAll])
```

Comment:
// Show usage: metadatacache.[drop|size|dropAll] ?

Reply:
Ok, will fix it.